libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2158,8 +2158,34 @@ void GCS_MAVLINK::send_local_position() const
         velocity.z);
 }
 
-void GCS_MAVLINK::send_planck_stateinfo() const
+void GCS::send_planck_stateinfo()
 {
+    for(uint8_t i=0; i<num_gcs(); i++) {
+        chan(i)->send_planck_stateinfo();
+    }
+}
+
+void GCS_MAVLINK::send_planck_stateinfo()
+{
+    //Sanity check
+    uint16_t now = AP_HAL::millis16();
+    if(now <= last_planck_stateinfo_sent_ms) {
+        last_planck_stateinfo_sent_ms = now;
+        return;
+    }
+
+    //Don't send if its too soon
```

Comment:
This is a common APM scheduling thing ?

Reply:
This by itself isn't, but it does make it such that the `STATEINFO` messages are sent out at whatever rate the stream was requested.  Normally, APM puts messages into "bins", and each bin is transmitted at some set rate.  They can then defer bins for transmission later, or reduce the rate for each bin (this happened to the STATEINFO messages when it got put into a bin).